### DISCUSSION TITLE
how to make v1.3.1 support reading virtual files for local attestation when sgx.remote_attestation = "none"

### DISCUSSION THREAD

**Discussion by sammyne** [gramineproject/gramine#1055] [Q&A]
It's ok for v1.2 to do local attestation without setting `sgx.remote_attestation`. However, after upgrade to v1.3.1, reading virtual file `/dev/attestation/my_target_info` would report file not found error. Really appreciate if someone can help me out of this problem.

**Comment by dimakuv** [comment]
Hm, this is a good point.

In v1.3.1, we introduced this explicit manifest option `sgx.remote_attestation = "none" | "epid" | "dcap"`.

And we introduced the logic of enabling pseudo-files like `my_target_info`, `report`, `user_report_data` **only** if `sgx.remote_attestation != "none"`. See this: https://github.com/gramineproject/gramine/blob/5bba58a902d099f9068e0f04b3c2a37e591a1d27/libos/src/fs/dev/attestation.c#L399-L403

But now I think that we were over-restrictive. If the underlying backend is Intel SGX, then local attestation is always accessible and doesn't care about/rely on EPID or DCAP infrastructure.

So you're right, I think these files should be accessible even if `sgx.remote_attestation = "none"`. Let me create a PR to fix this. Unfortunately, even if other maintainers agree to fix this problem, you'll have to wait for Gramine v1.4.

**Reply by sammyne** [reply]
Great thanks 😄 
Is it possible to release a v1.3.2 as hotfix? @dimakuv

**Reply by dimakuv** [reply]
> Is it possible to release a v1.3.2 as hotfix?

@sammyne Unfortunately not. This change itself is not crucial or security-related, so there is no point in releasing a new sub-version.

**Reply by sammyne** [reply]
> > Is it possible to release a v1.3.2 as hotfix?
> 
> @sammyne Unfortunately not. This change itself is not crucial or security-related, so there is no point in releasing a new sub-version.

Rendering local attestation broken should be a crucial bugs.

**Reply by dimakuv** [reply]
It's not a crucial bug for sure.

You can currently circumvent this by just having a dummy `sgx.remote_attestation = "dcap"` in your manifest file.

**Reply by sammyne** [reply]
> It's not a crucial bug for sure.
> 
> You can currently circumvent this by just having a dummy `sgx.remote_attestation = "dcap"` in your manifest file.

That would require me to setup PCCS stuff = =

**Reply by dimakuv** [reply]
@sammyne I don't think so, why?

**Reply by dimakuv** [reply]
https://github.com/gramineproject/gramine/pull/1057 -- I created a PR.

**Reply by boryspoplawski** [reply]
@sammyne I think @dimakuv is right and you could set `sgx.remote_attestation = whatever` without setting all that PCCS stuff.
Also if this fix (once merged) is so important you you, you can build Gramine manually
